zdm/grid.py

```diff
@@ -406,7 +421,16 @@ def GenMCFRB(self,pwb=None):
             only the relative probabilities of any given width.
             Hence, this routine only returns the integer of the width bin
             not the width itelf.
-        
+
+        Args:
+            pwb (optional): ??
```

Comment:
PWB stands for "probability(width,beam)". It is the total probability of detecting an FRB with a given width at a given beam sensitivity. It takes a lot to calculate, so having it stick in memory and only having it get recalculated once no matter how many FRBs are needed is very useful. This routine first decides which b, w to sample from. We could/should extend this memory to keeping a "rate" grid in memory for every single b,w - but this might be too memory intensive (O~5x5x1000x1000).

Reply:
updated